src/dsp_tools/commands/xmlupload/models/ingest.py

```diff
@@ -0,0 +1,91 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+from typing import Protocol
+
+import requests
+from loguru import logger
+from requests import Session
+from requests.adapters import HTTPAdapter
+from requests.adapters import Retry
+
+from dsp_tools.models.exceptions import BadCredentialsError
+from dsp_tools.models.exceptions import PermanentConnectionError
+from dsp_tools.models.exceptions import UserError
+from dsp_tools.utils.logger_config import logger_savepath
+
+
+@dataclass(frozen=True)
+class IngestResponse:
+    internal_filename: str
```

Comment:
Curiosity question: 
You use this class only once and only to retrieve the string. What is the advantage of wrapping that string in a class?

Reply:
I am used to type safety and it for me it is not uncommon to wrap a single `String` type in a wrapper. Also the actual response from the API contains more information. It would be easy to expand the class to incorporate more information if need may be.